rs/macros/core/src/shared.rs

```diff
@@ -223,3 +223,28 @@ fn replace_lifetime_with_static_in_path_args(path_args: PathArguments) -> PathAr
         path_args
     }
 }
+
+/// Check if type is `CommandResult<T>` and extract inner type `T`
+pub(crate) fn extract_result_type_with_value(ty: Type) -> (Type, bool) {
```

Comment:
Why do you need `bool` if you have `Option`. Can't you just have a single func which accepts `Type` and returns `Option<Type>`

Reply:
The type is always returned either original or unwrapped. See usage

Reply:
Eventually changed signature